src/pool/pool_jemalloc.c

```diff
@@ -0,0 +1,345 @@
+
+    if (ret != UMF_RESULT_SUCCESS) {
+        fprintf(stderr, "umfMemoryProviderFree failed \n");
+    }
+}
+
+static bool arena_extent_dalloc(extent_hooks_t *extent_hooks, void *addr,
+                                size_t size, bool committed,
+                                unsigned arena_ind) {
+    (void)extent_hooks; // unused
+    (void)addr;         // unused
+    (void)size;         // unused
+    (void)committed;    // unused
+    (void)arena_ind;    // unused
+
+    return true;
+}
+
+static bool arena_extent_commit(extent_hooks_t *extent_hooks, void *addr,
+                                size_t size, size_t offset, size_t length,
+                                unsigned arena_ind) {
+    (void)extent_hooks; // unused
+    (void)addr;         // unused
+    (void)size;         // unused
+    (void)offset;       // unused
+    (void)length;       // unused
+    (void)arena_ind;    // unused
+
+    return false;
+}
+
+static bool arena_extent_decommit(extent_hooks_t *extent_hooks, void *addr,
+                                  size_t size, size_t offset, size_t length,
+                                  unsigned arena_ind) {
+    (void)extent_hooks; // unused
+    (void)addr;         // unused
+    (void)size;         // unused
+    (void)offset;       // unused
+    (void)length;       // unused
+    (void)arena_ind;    // unused
+
+    return true;
+}
+
+static bool arena_extent_purge_lazy(extent_hooks_t *extent_hooks, void *addr,
+                                    size_t size, size_t offset, size_t length,
+                                    unsigned arena_ind) {
+    (void)extent_hooks; // unused
+    (void)size;         // unused
+
+    jemalloc_memory_pool_t *pool = get_kind_by_arena(arena_ind);
+    enum umf_result_t ret = umfMemoryProviderPurgeLazy(
+        pool->hProvider, (char *)addr + offset, length);
+
+    return (ret == UMF_RESULT_SUCCESS);
+}
+
+static bool arena_extent_purge_forced(extent_hooks_t *extent_hooks, void *addr,
+                                      size_t size, size_t offset, size_t length,
+                                      unsigned arena_ind) {
+    (void)extent_hooks; // unused
+    (void)size;         // unused
+
+    jemalloc_memory_pool_t *pool = get_kind_by_arena(arena_ind);
+    enum umf_result_t ret = umfMemoryProviderPurgeForce(
+        pool->hProvider, (char *)addr + offset, length);
+
+    return (ret == UMF_RESULT_SUCCESS);
+}
+
+static bool arena_extent_split(extent_hooks_t *extent_hooks, void *addr,
+                               size_t size, size_t size_a, size_t size_b,
+                               bool committed, unsigned arena_ind) {
+    (void)extent_hooks; // unused
+    (void)addr;         // unused
+    (void)size;         // unused
+    (void)size_a;       // unused
+    (void)size_b;       // unused
+    (void)committed;    // unused
+    (void)arena_ind;    // unused
+
+    return false;
```

Comment:
The same issue with return value. According to the documentation: "If the function returns true, this indicates that the extent remains unsplit and therefore should continue to be operated on as a whole."

Reply:
On Linux there's no need to split extents. The kernel splits them automatically. So this is correct (for reference, [here's](https://github.com/memkind/memkind/blob/master/src/memkind_arena.c#L270) the memkind implementation).

However, on Windows this requires additional calls. This is a gap in the provider abstraction - it needs these split/merge functions.

Reply:
@vinser52 Done